pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1118,13 +1118,24 @@ func maybeRetryBoot(ctx *domainContext, status *types.DomainStatus) {
 	}
 	defer file.Close()
 
+	if err := hyper.Task(status).VirtualTPMSetup(status.DomainName, agentName, ctx.ps, warningTime, errorTime); err != nil {
```

Comment:
Maybe I would simplify the code and put the `VirtualTPMSetup` call in the `doActivateTail` function
https://github.com/lf-edge/eve/blob/40fccb1ad15fef72934c464aaecd4a52226beece/pkg/pillar/cmd/domainmgr/domainmgr.go#L1747
`doActivateTail` is the function that starts the tasks, and it is the final state of both `doActivate` and `maybeRetryBoot`

Reply:
In that case if VirtualTPMSetup fails for some reason, the qemu config is now pointing to a socket that doesn't exist and vm start fails.

Reply:
Yep, you are right!